### ISSUE TITLE
Using LibraryImport instead of DllImport?

### ISSUE THREAD

**Issue by rmsimpson** [xoofx/Blake3.NET#27]
I ran a few benchmarks on my system, found the summaries interesting.  I rebuilt the code with .NET Core 9.0-Preview 2, then ran these benchmarks:
- Default with no changes
- Removing the SuppressGCTransition attribute
- Switching to LibraryImport
- Setting DisableRuntimeMarshalling attribute

Short version:  LibraryImport won handily on the 1,000,000 byte test

[Summary default.txt](https://github.com/xoofx/Blake3.NET/files/14728494/Summary.default.txt)
[Summary with GC.txt](https://github.com/xoofx/Blake3.NET/files/14728495/Summary.with.GC.txt)
[Summary with LibraryImport.txt](https://github.com/xoofx/Blake3.NET/files/14728496/Summary.with.LibraryImport.txt)
[Summary with LibraryImport_DisableRuntimeMarshalling.txt](https://github.com/xoofx/Blake3.NET/files/14728497/Summary.with.LibraryImport_DisableRuntimeMarshalling.txt)

**Comment by xoofx** [comment]
Yep, LibraryImport is much better, especially for string marshalling, PR welcome!
